pkg/appvalidation/business_case_test.go

```diff
@@ -127,9 +133,174 @@ func (s AppValidateTestSuite) TestBusinessCaseForUpdate() {
 		err := BusinessCaseForUpdate(&businessCase)
 		s.Error(err)
 		s.IsType(&apperrors.ValidationError{}, err)
-		expectedErrMessage := fmt.Sprintf("Could not validate *models.BusinessCase : " +
-			"{\"LifecycleCostPhase\":\"cannot have multiple costs for the same phase, solution, and year\"}",
+		expectedErrMessage := fmt.Sprintf("Could not validate *models.BusinessCase " + id.String() +
+			": {\"LifecycleCostPhase\":\"cannot have multiple costs for the same phase, solution, and year\"}",
 		)
 		s.Equal(expectedErrMessage, err.Error())
 	})
 }
+
+func (s AppValidateTestSuite) TestValidateAllRequiredLifecycleCosts() {
+	businessCase := testhelpers.NewBusinessCase()
+	dev := models.LifecycleCostPhaseDEVELOPMENT
+
+	s.Run("golden path", func() {
+		businessCase.LifecycleCostLines = testhelpers.NewValidLifecycleCosts(&businessCase.ID)
+		result := validateAllRequiredLifecycleCosts(&businessCase)
+		s.Equal(0, len(result))
+	})
+
+	s.Run("returns validations when there are missing years", func() {
+		businessCase.LifecycleCostLines = models.EstimatedLifecycleCosts{}
+		result := validateAllRequiredLifecycleCosts(&businessCase)
+		s.Equal("years 1, 2, 3, 4, 5 are required", result["asIsSolution"])
```

Comment:
Note from walkthrough: Add one more use case where we have at least 1 year, but not all of them. Don't just test the two ends of the valid spectrum.